### PR TITLE
Look for tokens passed through embed and handle them

### PR DESCRIPTION
**Pull request by jde**
## What does this PR do?

Allows auth tokens to be passed in through the embed string and used in all requests.

## How do I test this PR?

- Generate a token
- Pass it into the embed via the authToken param
- If the server accepts the token, you should be logged in like magic

### CODE REVIEW

File: views/article.ejs [hunk 1]
```diff
@@ -29,6 +29,7 @@
         talk: '/',
         asset_url: '<%= asset_url ? asset_url : '' %>',
         asset_id: '<%= asset_id ? asset_id : '' %>',
+        authToken: '',
```
Comment (alinhle) [inline views/article.ejs]: Should this be camelCase or snake_case?

File: client/coral-embed-stream/src/actions/config.js [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+export const addExternalConfig = (config) => {
+
+  /**
+   * If an authToken is present and not falsey, handle it.
+   *
+   * This allows the host page to pass in an authToken in order to control
+   * the Talk session.
+   */
+  if (config.authToken) {
+    handleAuthToken(config.authToken);
```
Comment (alinhle) [inline client/coral-embed-stream/src/actions/config.js]: To conform with the rest of the application, this should be dispatched.

**Review by alinhle — CHANGES_REQUESTED** [review]
I think we should avoid persisting the third party token. This token is generated and passed by the host site and usually it is already persistent somewhere, where the host site has access. This avoids having the token persisted at several places which is good for security. I'm  open for discussion.

**Comment by jde** [comment]
Closing in favor of approach here: https://github.com/coralproject/talk/pull/638

**Pull request closed without merging**